### ISSUE TITLE
about replace ce loss with similarity.

### ISSUE THREAD

**Issue by qianyunw** [KaiyangZhou/CoOp#15]
Hi Kaiyang,

thanks for you amazing work! 

I obtain.cross_entropy(output, label) is used in your training. I wonder if it is possible to replace it with similarity between text and image, by adding corresponding classname embedding in the the learned prompt. And do one-shot like clip. Is it possible to do so?

Thanks!

**Comment by KaiyangZhou** [comment]
the input to the cross-entropy is computed based on similarity scores between image and text features (the latter use class names in the prompts)

maybe I misunderstand your question?

**Comment by qianyunw** [comment]
谢谢，王倩芸已收到。
